homeassistant/components/sky_hub/device_tracker.py

```diff
@@ -1,63 +1,71 @@
 """Support for Sky Hub."""
 import logging
-import re
 
-import requests
+from pyskyqhub.skyq_hub import SkyQHub
 import voluptuous as vol
 
 from homeassistant.components.device_tracker import (
     DOMAIN,
     PLATFORM_SCHEMA,
     DeviceScanner,
 )
-from homeassistant.const import CONF_HOST, HTTP_OK
+from homeassistant.const import CONF_HOST
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
-_MAC_REGEX = re.compile(r"(([0-9A-Fa-f]{1,2}\:){5}[0-9A-Fa-f]{1,2})")
+_ERROR = logging.ERROR
+_INFO = logging.INFO
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Optional(CONF_HOST): cv.string})
 
 
-def get_scanner(hass, config):
+async def async_get_scanner(hass, config):
     """Return a Sky Hub scanner if successful."""
-    scanner = SkyHubDeviceScanner(config[DOMAIN])
-
+    host = config[DOMAIN].get(CONF_HOST, "192.168.1.254")
+    websession = async_get_clientsession(hass)
+    hub = SkyQHub(websession, host)
+    scanner = SkyHubDeviceScanner(config[DOMAIN], hub, host)
+    await scanner.async_connect(hass)
     return scanner if scanner.success_init else None
 
 
 class SkyHubDeviceScanner(DeviceScanner):
     """This class queries a Sky Hub router."""
 
-    def __init__(self, config):
+    def __init__(self, config, hub, host):
         """Initialise the scanner."""
-        _LOGGER.info("Initialising Sky Hub")
-        self.host = config.get(CONF_HOST, "192.168.1.254")
+        self._connection_failed = False
+        self._dataparse_failed = False
+        self._hub = hub
+        self.host = host
         self.last_results = {}
-        self.url = f"http://{self.host}/"
+        self.success_init = False
 
-        # Test the router is accessible
-        data = _get_skyhub_data(self.url)
-        self.success_init = data is not None
+    async def async_connect(self, hass):
+        """Test the router is accessible."""
+        _LOGGER.info("Initialising Sky Hub")
+        await self._hub.async_connect()
+        self.success_init = self._hub.success_init
 
-    def scan_devices(self):
+    async def async_scan_devices(self):
         """Scan for new devices and return a list with found device IDs."""
-        self._update_info()
+        await self._async_update_info()
 
         return (device for device in self.last_results)
```

Comment:
We should return a list of mac address strings. Currently we're returning a generator.

Reply:
I've not changed that line, have the code requirements changed?